docs/tutorial.rst

```diff
@@ -1,4 +1,209 @@
 Tutorial
 ========
 
-*Coming soon!*
+In this document, we are going to develop an application that will
+convert a given time (and date) between two time zones.
+The user will enter a date and a time,
+and select two time zones from a list of all available time zones,
+one for the source location and one for the destination location.
+
+.. note::
+
+   Time zones are represented in "region/location" format,
+   as in "Australia/Tasmania".
+   While most such codes have two components,
+   some contain only one (like "UTC"),
+   and some contain more than two (like "America/North_Dakota/New_Salem".
+   Also note that spaces in region and location names are replaced
+   with underscores.
+   See the `Wikipedia list`_ for more information.
+
+
+Prerequisites
+-------------
+
+First of all, we have to install the dependencies.
+These are: *clastic* (obviously) and *dateutil*.
+Note that the PyPI package name for *dateutil* is *python-dateutil*::
+
+  pip install clastic python-dateutil
+
+
+
+Getting started
+---------------
+
+Here is the initial version of our application that only displays
+the form but doesn't handle the submitted data:
+
+.. code-block:: python
+
+   import os
+
+   from clastic import Application
+   from clastic.render import AshesRenderFactory
+   from dateutil import zoneinfo
+
+
+   def home():
+       zone_info = zoneinfo.get_zonefile_instance()
+       zone_names = zone_info.zones.keys()
+       entries = ((zone.split("/")[-1], zone) for zone in zone_names)
+       zones = [
+           {"location": location.replace("_", " "), "zone": zone}
+           for location, zone in sorted(entries)
+       ]
+       return {"zones": zones}
+
+
+   def create_app():
+       routes = [("/", home, "home.html")]
+       cur_path = os.path.dirname(__file__)
```

Comment:
I tend to move `cur_path` up to the module level as a global `CUR_PATH` for simplicity.

Reply:
But it's not used anywhere else. Is it worth introducing a global variable for this? But I don't feel very strongly either way; if you prefer it that way, I'm OK,

We could skip the variable definition and give the expression directly as parameter to the render factory. Would that hurt readability?

Reply:
Ah, I'm not worried about an extra line. Mostly, I've learned from experience that it's best to do

```python
CUR_PATH = os.path.dirname(os.path.abspath(__file__))
```

at the module level. Otherwise the path could be relative and/or nterpreted differently between function calls. This way it gets set once, at import time. Not a big deal, just a best practice in my experience.

Reply:
OK, done.